source/geod24/concurrency.d

```diff
@@ -1,2583 +1,1608 @@
-
-// Generator
-
-/**
- * If the caller is a Fiber and is not a Generator, this function will call
- * scheduler.yield() or Fiber.yield(), as appropriate.
- */
-void yield() nothrow
-{
-    auto fiber = Fiber.getThis();
-    if (!(cast(IsGenerator) fiber))
+    /***************************************************************************
+
+        This created a new Fiber for the supplied op and adds it to the
+        dispatch list.
+
+    ***************************************************************************/
+
+    public void spawn (void delegate() op, size_t sz=0) nothrow
     {
-        if (scheduler is null)
-        {
-            if (fiber)
-                return Fiber.yield();
-        }
-        else
-            scheduler.yield();
+        create(op, sz);
+        yield();
     }
-}
 
-/// Used to determine whether a Generator is running.
-private interface IsGenerator {}
 
+    /***************************************************************************
 
-/**
- * A Generator is a Fiber that periodically returns values of type T to the
- * caller via yield.  This is represented as an InputRange.
- */
-class Generator(T) :
-    Fiber, IsGenerator, InputRange!T
-{
-    /**
-     * Initializes a generator object which is associated with a static
-     * D function.  The function will be called once to prepare the range
-     * for iteration.
-     *
-     * Params:
-     *  fn = The fiber function.
-     *
-     * In:
-     *  fn must not be null.
-     */
-    this(void function() fn)
+        If the caller is a scheduled Fiber, this yields execution to another
+        scheduled Fiber.
+
+    ***************************************************************************/
+
+    public void yield () nothrow
     {
-        super(fn);
-        call();
+        // NOTE: It's possible that we should test whether the calling Fiber
+        //       is an InfoFiber before yielding, but I think it's reasonable
+        //       that any (non-Generator) fiber should yield here.
+        if (Fiber.getThis())
+            Fiber.yield();
     }
 
-    /**
-     * Initializes a generator object which is associated with a static
-     * D function.  The function will be called once to prepare the range
-     * for iteration.
-     *
-     * Params:
-     *  fn = The fiber function.
-     *  sz = The stack size for this fiber.
-     *
-     * In:
-     *  fn must not be null.
-     */
-    this(void function() fn, size_t sz)
+
+    /***************************************************************************
+
+        Returns an appropriate ThreadInfo instance.
+
+        Returns a ThreadInfo instance specific to the calling Fiber if the
+        Fiber was created by this dispatcher, otherwise it returns
+        ThreadInfo.thisInfo.
+
+    ***************************************************************************/
+
+    public @property ref ThreadInfo thisInfo () nothrow
     {
-        super(fn, sz);
-        call();
+        auto f = 
```

Comment:
What is a `trancontexter`? I have never seen this word before.

Reply:
Thank you for your review.